tools/occlum-trace/src/main.rs

```diff
@@ -0,0 +1,768 @@
+use std::collections::HashSet;
+use std::fs::File;
+use std::io::Write;
+use std::str::FromStr;
+use std::{collections::HashMap, io::Read, fs};
+use std::path::PathBuf;
+
+use clap::{App, Arg, ArgMatches};
+use regex::Regex;
+
+
+#[macro_use]
+extern crate lazy_static;
+
+
+static TIME_PATTERN: &'static str = r"\[(?P<time>(\d{4})-(\d{2})-(\d{2})T(\d{2}):(\d{2}):(\d{2})\.(\d{3})Z)\]";
+static LOG_LEVEL_PATTERN: &'static str = r"\[(TRACE|DEBUG|ERROR| INFO| WARN)\]";
+static THREAD_PATTERN: &'static str = r"\[T(?P<thread_id>\d{1,})\]";
+static SYSCALL_NUMBER_PATTERN: &'static str = r"\[#(?P<syscall_number>\d{1,})\]";
+static SYSCALL_NAME_PATTERN: &'static str = r"\[·*(?P<syscall_name>[a-zA-Z]{1,}[0-9]*)\]";
+static SYSCALL_ENTRY_PATTERN: &'static str = r"Syscall \{(?P<syscall_name> num = [a-zA-Z]+[0-9]*,?) ";
+static SYSCALL_RETURN_PATTERN: &'static str = r"Retval = (?P<return_value>-?\d+)";
+
+static DECORATOR_PATTERN: &'static str = r"\u{1b}\[\d+m";
+
+//check_open
+static OPEN_AT_ABSOLUTE_PATTERN: &'static str = r"openat: fs_path: FsPath \{ (Absolute|CwdRelative)\((?P<file_path>.+)\) \},";
+static OPEN_AT_RELATIVE_PATTERN: &'static str = r"openat: fs_path: FsPath \{ FdRelative\((?P<file_path>.+), \d+\) \}";
+static FILENAME_PATTERN: &'static str = r".*/(?P<file_name>.*)$";
+
+lazy_static! {
+    static ref TIME_REGEX: Regex = {
+        Regex::new(TIME_PATTERN).unwrap()
+    };
+    static ref LOG_LEVEL_REGEX: Regex = {
+        Regex::new(LOG_LEVEL_PATTERN).unwrap()
+    };
+    static ref THREAD_REGEX: Regex = {
+        Regex::new(THREAD_PATTERN).unwrap()
+    };
+    static ref SYSCALL_NUMBER_REGEX: Regex = {
+        Regex::new(SYSCALL_NUMBER_PATTERN).unwrap()
+    };
+    static ref SYSCALL_NAME_REGEX: Regex = {
+        Regex::new(SYSCALL_NAME_PATTERN).unwrap()
+    };
+    static ref DECORATOR_REGEX: Regex = {
+        Regex::new(DECORATOR_PATTERN).unwrap()
+    };
+    static ref SYSCALL_ENTRY_REGEX: Regex = {
+        Regex::new(SYSCALL_ENTRY_PATTERN).unwrap()
+    };
+    static ref SYSCALL_RETURN_REGEX: Regex = {
+        Regex::new(SYSCALL_RETURN_PATTERN).unwrap()
+    };
+}
+
+//check open
+lazy_static! {
+    static ref OPEN_AT_ABSOLUTE_REGEX: Regex = {
+        Regex::new(OPEN_AT_ABSOLUTE_PATTERN).unwrap()
+    };
+    static ref OPEN_AT_RELATIVE_REGEX: Regex = {
+        Regex::new(OPEN_AT_RELATIVE_PATTERN).unwrap()
+    };
+    static ref FILENAME_REGEX: Regex = {
+        Regex::new(FILENAME_PATTERN).unwrap()
+    };
+}
+
+
+#[derive(Debug, Clone)]
+struct TraceOption {
```

Comment:
Occlum is using clap in other tools. Maybe we could use the same crate in this tool.

Reply:
Fixed.